src/consent.ts

```diff
@@ -1,9 +1,131 @@
-export default function Consent(mpInstance) {
-    var self = this;
-    var CCPAPurpose = 'data_sale_opt_out';
+import {
+    CCPAConsentState,
+    ConsentState,
+    GDPRConsentState,
+    PrivacyConsentState,
+} from '@mparticle/web-sdk';
+import { MParticleUser, MParticleWebSDK } from './sdkRuntimeModels';
+import { Dictionary, isObject } from './utils';
 
-    // this function is called when consent is required to determine if a cookie sync should happen, or a forwarder should be initialized
-    this.isEnabledForUserConsent = function(consentRules, user) {
+export const CCPAPurpose = 'data_sale_opt_out' as const;
```

Comment:
Is the as const necessary?

Reply:
I kind of is. the `as const` tells the compiler that this value is `readonly` and lets us use it in line 15 without issues.

Reply:
maybe i should have been more clear - is saying **const** CCPAPurpose = .... **as const** redundant?

Reply:
It looks like that, but `const` at the beginning of the statement versus at the end signify two different things.

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions